geoproxy/main.go

```diff
@@ -272,87 +315,230 @@ func getSubmitProxyUrl(rawCoord []float64) string {
+	buff := bytes.NewBuffer(body)
+	bodyStr := buff.String()
+	// This is where we want to insert the id param	
+	i := strings.LastIndex(bodyStr, "}")
+	// Convert to runes to split
+	runes := []rune(bodyStr)
+	// Left side string of \n}
+	leftStr := string(runes[0:i])
+	buff = bytes.NewBufferString(leftStr)
+	// Concatenate new id param and request body remainder
+	buff.WriteString(",\"id\":" + "\"" + id.String() + "\"")
+	buff.WriteString(string(runes[i:len(runes)]))
+	buffBytes := buff.Bytes()	
+	
+	setupTimer(id)
+	//fmt.Println(buff.String())
+	if verbose {
+		log.Printf("body %s \n",string(buffBytes))
+	}
+	// Send to other servers for view request
+	for i := 0; i < len(target); i++ {
+		// parse the url
+		url, _ := url.Parse(target[i])
+		url.Path = "/view"
+		if verbose {
+			log.Printf("url: %s \n", target[i])
+		}
+		newReqBody := ioutil.NopCloser(bytes.NewBuffer(buffBytes))
+		// reusing requests is unreliable, so copy to new request
+
+		/*
+		res, err := http.DefaultClient.Do(newReq)
+		*/
+		res, err := http.Post(url.String(), "application/json", newReqBody)
+		// Todo: Question: is it alright if we make the request like this? Or do we have to copy over the headers and stuff like before?
+		// Also is Post preferable? or is there not much of a difference?
+		if err != nil {
+			log.Printf("Error when sending request", err)
+		} else {
+			defer res.Body.Close()
+			var responseObj ResponseObj
+			/*
+			fmt.Println("HTTP Response Status:", res.StatusCode, http.StatusText(res.StatusCode))
+			//fmt.Println(res.Body)
+			buf := new(bytes.Buffer)
+			buf.ReadFrom(res.Body)
+			newStr:=buf.String()
+			fmt.Println(newStr)
+		
+			if verbose {
+				for k, v := range res.Header {
+					log.Printf("head: %s, val: %s \n", k, v)
+				}
+			}
+			*/
+			body, bodyErr := ioutil.ReadAll(res.Body)
+			if bodyErr != nil {
+				// Do we want to stop the program here 
+				log.Fatal(bodyErr)
+			}
 
-	// Note that ServeHttp is non blocking and uses a go routine under the hood
-	proxy.ServeHTTP(res, req)
+			if unmarshalErr := json.Unmarshal([]byte(body), &responseObj); unmarshalErr != nil {
+				log.Fatal(unmarshalErr)
+			}
+			processResponse(responseObj)
+			if verbose {
+				log.Printf("Request served to reverse proxy for %s\n", target[i])
+				log.Printf("Response obj: %v", responseObj)
+			}
+		}
+	}
 }
 
 // Enable cors for response to pre-flight
 func enableCors(w *http.ResponseWriter) {
 	//  allow CORS
 	(*w).Header().Set("Access-Control-Allow-Origin", "*")
-
 	//  Allow these headers in client's response to pre-flight response
 	(*w).Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
 	(*w).Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
 }
 
+// Sets up timeout callback. Call after sending all requests
+func setupTimer(id uuid.UUID) {
+	// How long we should wait for multiple
+	timeWait := 4 * time.Second
```

Comment:
TODO: Possibly set as global const or change according to how many view requests are being sent